src/emitter/diff.rs

```diff
@@ -36,6 +36,13 @@ impl Emitter for DiffEmitter {
                     &self.config,
                 );
             }
+        } else if original_text != formatted_text {
+            // This occurs when the only difference between the original and formatted values
+            // is the newline style. This happens because The make_diff function compares the
+            // original and formatted values line by line, independent of line endings.
+            let file_path = ensure_real_path(filename);
+            writeln!(output, "Incorrect newline style in {}", file_path.display())?;
+            return Ok(EmitterResult { has_diff: true });
         }
```

Comment:
When running in `--check` mode, this is the emitter used, and like many of the other emitters, it leverages the `make_diff` function in [rustfmt_diff.rs](https://github.com/rust-lang/rustfmt/blob/master/src/rustfmt_diff.rs#L183).

That function internally leverages the [diff](https://crates.io/crates/diff) crate check for any formatting changes on a line by line basis. However, internally `diff` [leverages](https://github.com/utkarshkukreti/diff.rs/blob/master/src/lib.rs#L17-L35) the [str.lines()](https://doc.rust-lang.org/std/primitive.str.html#method.lines) function which ignores line endings. 

This ensures that `rustfmt --check ..` will exit with a non-zero exit code, even if the only change `rustfmt` would make is the newline style. I added a line to emit the name of the file if there was a newline style issue, because IMO it would be confusing if the only issue in my project was newline style and running `rustfmt --check` was giving me a non-zero exit code (due to the newline style diffs) without giving me any indication why.

Reply:
If `make_diff` use [str.split](https://doc.rust-lang.org/std/primitive.str.html#method.split) then this additional check wouldn't be needed ? The error message could be less clear though...

Reply:
Thanks @scampi! 

> If make_diff use str.split then this additional check wouldn't be needed ? The error message could be less clear though...

Just to clarify, are you asking if we could avoid this check here if we updated `make_diff` to include checking line endings? If so, then potentially yes, although that would result in the line ending characters always being displayed in the output when there's any diff. Also in the instance of newline style only diffs, I think it would most often result in `rustfmt` printing the entire contents of the file twice (for example if the original used `\r\n` and `rustfmt` changed the newline style to `\n`).

I could take a look into what would be involved to make that happen in `make_diff` if you'd like?